src/Microsoft.AspNet.Hosting/Internal/HostingEngine.cs

```diff
@@ -70,14 +70,16 @@ public virtual IDisposable Start()
             var server = ServerFactory.Start(_serverInstance,
                 async features =>
                 {
-                    var httpContext = contextFactory.CreateHttpContext(features);
-                    httpContext.ApplicationServices = _applicationServices;
-                    var requestIdentifier = GetRequestIdentifier(httpContext);
-
-                    using (logger.BeginScope("Request Id: {RequestId}", requestIdentifier))
+                    using (var httpContext = contextFactory.CreateHttpContext(features))
```

Comment:
Isn't FireOnCompleted/OnCompleted called by the end of ```await application(httpContext);```?

e.g. https://github.com/aspnet/KestrelHttpServer/blob/dev/src/Microsoft.AspNet.Server.Kestrel/Http/Frame.cs#L226

Else it will never be called which would be a problem?